src/premiscale/hypervisor/_base.py

```diff
@@ -69,4 +90,24 @@ def close(self) -> None:
         """
         Close the connection with the Libvirt hypervisor.
         """
-        self.connection.close()
\ No newline at end of file
+        if self._connection:
+            self._connection.virConnectClose()
+            log.info(f'Closed connection to host at {self.connection_string}')
+        else:
+            log.error(f'No host connection to close, probably due to an error on connection open.')
+
+    def _configure_ssh(self) -> None:
+        """
+        Configure the SSH connection to the host. This method makes connection timeouts configurable.
+        """
+        ssh_config_p = SSHConfig.from_path(os.path.expanduser('~/.ssh/config'))
```

Comment:
This should be reworked at some point; it's a large dependency for not that much value add.

Reply:
I removed this dependency and put together my own basic solution.